cmd/clusterctl/client/config.go

```diff
@@ -302,16 +303,40 @@ func (c *clusterctlClient) templateOptionsToVariables(options GetClusterTemplate
 	}
 
 	// the ControlPlaneMachineCount, if valid, can be used in templates using the ${ CONTROL_PLANE_MACHINE_COUNT } variable.
-	if options.ControlPlaneMachineCount < 1 {
+	if options.ControlPlaneMachineCount == nil {
+		//Check if set through env variable and default to 1 otherwise
+		if v, err := c.configClient.Variables().Get("CONTROL_PLANE_MACHINE_COUNT"); err != nil {
+			options.ControlPlaneMachineCount = utilpointer.Int32Ptr(1)
+		} else {
+			i, err := strconv.Atoi(v)
```

Comment:
Minor: If we switch `ControlPlaneMachineCount` and `WorkerMachineCount` to be `*int64` instead, we can use https://golang.org/pkg/strconv/#ParseInt instead of `Atoi`